SentryTestUtils/TestSentryNSTimerFactory.swift

```diff
@@ -1,52 +1,92 @@
         var invocationInfo: InvocationInfo?
+        
+        init(timer: Timer, interval: TimeInterval? = nil, block: ((Timer) -> Void)? = nil, lastFireDate: Date, invocationInfo: InvocationInfo? = nil) {
+            self.timer = timer
+            self._interval = interval
+            self.block = block
+            self.lastFireDate = lastFireDate
+            self.invocationInfo = invocationInfo
+        }
     }
+    
+    public var overrides: Overrides?
+    
+    private var currentDateProvider: SentryCurrentDateProvider
+    
+    public init(currentDateProvider: SentryCurrentDateProvider) {
+        self.currentDateProvider = currentDateProvider
+        super.init()
+    }
+}
 
-    public var overrides = Overrides()
-
-    public override func scheduledTimer(withTimeInterval interval: TimeInterval, repeats: Bool, block: @escaping (Timer) -> Void) -> Timer {
+// MARK: Superclass overrides
+public extension TestSentryNSTimerFactory {
+    override func scheduledTimer(withTimeInterval interval: TimeInterval, repeats: Bool, block: @escaping (Timer) -> Void) -> Timer {
         let timer = Timer.scheduledTimer(withTimeInterval: TimeInterval.infinity, repeats: repeats, block: block)
-        overrides.timer = timer
-        overrides.block = block
+        overrides = Overrides(timer: timer, interval: interval, block: block, lastFireDate: currentDateProvider.date())
         return timer
     }
-
-    public override func scheduledTimer(withTimeInterval ti: TimeInterval, target aTarget: Any, selector aSelector: Selector, userInfo: Any?, repeats yesOrNo: Bool) -> Timer {
+    
+    override func scheduledTimer(withTimeInterval ti: TimeInterval, target aTarget: Any, selector aSelector: Selector, userInfo: Any?, repeats yesOrNo: Bool) -> Timer {
         let timer = Timer.scheduledTimer(timeInterval: ti, target: aTarget, selector: aSelector, userInfo: userInfo, repeats: yesOrNo)
         //swiftlint:disable force_cast
-        overrides.invocationInfo = Overrides.InvocationInfo(target: aTarget as! NSObject, selector: aSelector)
+        let invocationInfo = Overrides.InvocationInfo(target: aTarget as! NSObject, selector: aSelector)
         //swiftlint:enable force_cast
+        overrides = Overrides(timer: timer, interval: ti, lastFireDate: currentDateProvider.date(), invocationInfo: invocationInfo)
         return timer
     }
+}
+
+// MARK: Extensions
+public extension TestSentryNSTimerFactory {
+    enum TestTimerError: Error {
+        case timerNotInitialized
+    }
+    
+    // check the current time against the last fire time and interval, and if enough time has elapsed, execute any block/invocation registered with the timer
+    func check() throws {
+        guard let overrides = overrides else { throw TestTimerError.timerNotInitialized }
+        let currentDate = currentDateProvider.date()
+        if currentDate.timeIntervalSince(overrides.lastFireDate) >= overrides.interval {
+            try fire()
+        }
+    }
```

Comment:
This function is a stronger assertion than just firing the timer block directly. Now, we're actually testing that the timer fires after the correct interval. I think this lead to some false confidence in tests that would call `fire` and sort of throw away the context of how much time has elapsed or what should happen. See comments in SentryTracerTests below.